saleor/cart/decorators.py

```diff
@@ -1,85 +1,120 @@
 from __future__ import unicode_literals
 from functools import wraps
 
+from django.utils.timezone import now
+
 from .core import set_cart_cookie
 from .models import Cart
 
 
-def get_user_open_cart_token(user):
-    cart = Cart.get_user_open_cart(user)
-    if cart is not None:
-        return cart.token
+def find_and_assign_cart(request, queryset=Cart.objects.all()):
+    """Assign cart from cookie to request user
+    :type request: django.http.HttpRequest
+    """
+    token = request.get_signed_cookie(Cart.COOKIE_NAME, default=None)
+    if not token:
+        return
+    cart = get_anonymous_cart_from_token(token=token, cart_queryset=queryset)
```

Comment:
What if the card exists but is not anonymous? Say I log out and log in as a different user?